internal/ethapi/api.go

```diff
@@ -773,6 +787,37 @@ func DoCall(ctx context.Context, b Backend, args CallArgs, blockNr rpc.BlockNumb
 	} else {
 		addr = *args.From
 	}
+	// Override the fields of specified contracts before execution.
+	if accounts != nil {
```

Comment:
AFAIK you don't need to check for `nil` here, you can range over a nil map and it will correctly noop.